cmake: Don't override BLAS/LAPACK linker flags.

mmuetzel opened the pull request:
When OpenBLAS is the only BLAS/LAPACK that is installed on the build system, the static linker flags in the pkg-config files might be wrong.
They contain the following line:
```
Libs.private: BLAS::BLAS
```
It should be:
```
Libs.private: -lopenblas
```

During the configuration step, the following is printed:
```
   -- BLAS:
      -- link:    F:/msys64/ucrt64/lib/libopenblas.dll.a
   -- LAPACK:
      -- link:    BLAS::BLAS
```

That seems to be caused by the following command in the `CMakeLists.txt` file:
```
get_target_property(LAPACK_LIBRARIES LAPACK::LAPACK INTERFACE_LINK_LIBRARIES) # Get variables from target (*.pc/cmake, msg).
```

I'm not sure if this is a cmake bug or if this is the expected result.

Anyway, the value of `LAPACK_LIBRARIES` seems to be set correctly immediately before that command. Simply removing it fixes the content of the `.pc` file.

For consistency, this PR also removes the corresponding command for the `BLAS_LIBRARIES` variable.

See also: https://github.com/msys2/MINGW-packages/issues/11376

fghoussen commented:
Thanks for reporting! Sounds relevant.
Can you kill all occurrences of `get_target_property`? If relevant for blas/lapack, same should apply to other dependencies (MPI, eigen, ...), no? Also, change you update the change log?

mmuetzel commented:
Thanks for the review.

I removed all occurrences of `get_target_property` from `CMakeLists.txt`. That seems to work correctly for me. The content of the generated `.pc` and `.cmake` files look good to me after this change.

I also added a note to the `CHANGES` file.
